db/migrate/20190118204247_move_miq_set_memberships.rb

```diff
@@ -0,0 +1,38 @@
+require 'ancestry'
+
+class MoveMiqSetMemberships < ActiveRecord::Migration[5.0]
+  class Relationship < ActiveRecord::Base
+    has_ancestry
+  end
+
+  class MiqSetMembership < ActiveRecord::Base
+  end
+
+  def up
+    say_with_time("Converting set relationships to miq_set_memberships") do
+      # ancestry holds the id for the relationship record that references the set
+      # the parent record contains the set object
+      # because these are trivial trees, we can look up the parent and get the resource_id
+      parent_miq_set_id = "(SELECT rel2.resource_id FROM relationships rel2 WHERE rel2.id = rels.ancestry::bigint)"
+
+      connection.execute(<<-SQL)
+        INSERT INTO miq_set_memberships (member_type, miq_set_id, member_id, created_at, updated_at)
+          SELECT rels.resource_type, #{parent_miq_set_id}, rels.resource_id, rels.created_at, rels.updated_at
+          FROM relationships rels
+          WHERE rels.relationship = 'membership' AND rels.ancestry IS NOT NULL
+      SQL
+
+      Relationship.where(:relationship => "membership").delete_all
+    end
+  end
+
+  def down
+    say_with_time("Converting miq_set_memberships to set relationships") do
+      MiqSetMembership.all.each do |m|
```

Comment:
ok. So according to above, we need relationship records for the sets and the membership records.

Think we need one loop for each of the sets. (probably store the set_id to relationship record id in a hash for the next step)

Then we can run the membership loop and the new relationship will have the set as the parent / ancestry value.

would be cool if the tests mimic each other in this way